packages/in_app_purchase/in_app_purchase_ios/test/in_app_purchase_ios_platform_test.dart

```diff
@@ -273,6 +273,60 @@ void main() {
       expect(completerError.message, 'ios_domain');
       expect(completerError.details, {'message': 'an error message'});
     });
+    test(
+        'should get canceled purchase status when error code is SKErrorPaymentCancelled',
+        () async {
+      fakeIOSPlatform.testTransactionCancel = 2;
+      List<PurchaseDetails> details = [];
+      Completer completer = Completer();
+
+      Stream<List<PurchaseDetails>> stream = iapIosPlatform.purchaseStream;
+      late StreamSubscription subscription;
+      subscription = stream.listen((purchaseDetailsList) {
+        details.addAll(purchaseDetailsList);
+        purchaseDetailsList.forEach((purchaseDetails) {
+          if (purchaseDetails.status == PurchaseStatus.canceled) {
+            completer.complete(purchaseDetails.status);
+            subscription.cancel();
+          }
+        });
+      });
+      final AppStorePurchaseParam purchaseParam = AppStorePurchaseParam(
+          productDetails:
+              AppStoreProductDetails.fromSKProduct(dummyProductWrapper),
+          applicationUserName: 'appName');
+      await iapIosPlatform.buyNonConsumable(purchaseParam: purchaseParam);
+
+      PurchaseStatus purchaseStatus = await completer.future;
+      expect(purchaseStatus, PurchaseStatus.canceled);
+    });
+    test(
```

Comment:
nit: could use an empty line to make it clear a new test starts here:
```suggestion

    test(
```